docs/yul.rst

```diff
@@ -952,6 +952,20 @@ option.
 
 See :ref:`Using the Commandline Compiler <commandline-compiler>` for details about the Solidity linker.
 
+memoryguard
+^^^^^^^^^^^
+
+This function is available in the EVM dialect with objects. The caller of
+``let ptr := memoryguard(size)`` promises that they only use memory in either
```

Comment:
Does `size` need to be a literal number?

Reply:
Added it right there now.